addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -2718,12 +2922,88 @@ const Wysiwyg = Widget.extend({
                 firstChild.setAttribute('data-last-history-steps', historyIds);
             }
         }
-    },
+    }
     _bindOnBlur() {
         this.$editable.on('blur', this._onBlur);
-    },
-
-});
+    }
+    _trigger_up(ev) {
+        const evType = ev.name;
+        const payload = ev.data;
+        if (evType === 'call_service') {
+            this._callService(payload);
+        } else if (evType === 'get_session') {
+            if (payload.callback) {
+                payload.callback(this.env.session);
+            }
+        } else if (evType === 'load_views') {
+            const params = {
+                model: payload.modelName,
+                context: payload.context,
+                views_descr: payload.views,
+            };
+            this.env.dataManager
+                .load_views(params, payload.options || {})
+                .then(payload.on_success);
+        } else if (evType === 'load_filters') {
+            return this.env.dataManager
+                .load_filters(payload)
+                .then(payload.on_success);
+        }
+    }
```

Comment:
Wait that is a new function ? Is it like copy pasted from the legacy code ? I'm confused... same for below.